omnigibson/systems/macro_particle_system.py

```diff
@@ -905,12 +1011,15 @@ def _deserialize(cls, state):
             obj_uuid, n_particles = int(state[idx]), int(state[idx + 1])
             obj = og.sim.scene.object_registry("uuid", obj_uuid)
             group_obj_id2link = {i: link_name for i, link_name in enumerate(obj.links.keys())}
+            # # Parse -1 when there's no link
+            # group_obj_id2link[-1] = None
             group_objs.append(obj)
             groups_dict[obj.name] = dict(
                 particle_attached_obj_uuid=obj_uuid,
                 n_particles=n_particles,
                 particle_idns=[int(idn) for idn in state[idx + 2 : idx + 2 + n_particles]], # Idx + 2 because the first two are obj_uuid and n_particles
-                particle_attached_link_names=[group_obj_id2link[int(idn)] for idn in state[idx + 2 + n_particles : idx + 2 + n_particles * 2]],
+                particle_attached_link_names=[group_obj_id2link[int(idn)] if idn > 0 else int(idn)
+                                              for idn in state[idx + 2 + n_particles : idx + 2 + n_particles * 2]],
```

Comment:
can we avoid confounding links and faces?

Reply:
done